Gordon360/Services/AccountService.cs

```diff
@@ -109,6 +113,9 @@ public AccountViewModel GetAccountByUsername(string username)
             string? minor,
             string? hall,
             string? classType,
+            string? preferredClassYear,
+            string? initialYear,
+            string? finalYear,
```

Comment:
@EjPlatzer is this a viable solution? I originally thought a viewmodel would be possible but is relatively difficult with the implementation using every specific search `byquery`. In my mind we _could_ put the viewmodel as an string in the uri, but that seems like bad practice. Otherwise, I'm stumped on a cleaner solution.

Reply:
What is the goal of having a view model instead of separate fields? I see from the logic below that you require both initial and final year to be present for the search to work, so you could enforce that with a view model.

You definitely can include a view model in the search params. For example, if you have the model
```csharp
public record ClassYearRange(int Initial, int Final)
```
you would end up with `ClassYearRange.Initial=2020&ClassYearRange.Final=2022` in the query string. I haven't done that before so I don't know exactly how you'd implement it on the frontend but it shouldn't be hard. That way, you could make the model optional but the fields within the model not optional, meaning the binding would fail if only one of the fields is present.

That's potentially a lot of extra work for some validation that may not be necessary. I don't have a strong opinion either way.